### PR TITLE
[MRG] Raise NotImplementedError for vectorized Brainvision ASCII data file

### LINKED ISSUE #5142: mne.io.read_raw_brainvision reads incorrect number of datapoints
I am writing a simple time series visualization script basing on MNE. I need to read data from EDF and BrainVision txt files, I have a sample that was exported to both. I discovered a strange behaviour that probably is a bug in `mne.io.read_raw_brainvision`, it is illustrated by

```
#!/usr/bin/env python
import mne

edf = mne.io.read_raw_edf("datasets/example.edf")
bv = mne.io.read_raw_brainvision("datasets/dattxt/example.vhdr")

print("Number of datapoints per channel ", edf.n_times, bv.n_times)
```

which prints in my case `('Number of datapoints per channel ', 831488, 73)`. It can be found (together with sample data) [in this repo](https://github.com/ociepkam/pfb). I would expect that `edf.n_times` and `bv.n_times` be equal. 

Additionally, when I try to read data from `bv` like so

```
data, time = bv[:, :1000]
```

I get

```
Traceback (most recent call last):
  File "./test.py", line 9, in <module>
    data, time = bv[:, :1000]
  File "/home/arccha/.conda/envs/pfb/lib/python2.7/site-packages/mne/io/base.py", line 841, in __getitem__
    verbose=self.verbose)
  File "/home/arccha/.conda/envs/pfb/lib/python2.7/site-packages/mne/io/base.py", line 523, in _read_segment
    cals, mult)
  File "/home/arccha/.conda/envs/pfb/lib/python2.7/site-packages/mne/io/brainvision/brainvision.py", line 116, in _read_segment_file
    _read_segments_c(self, data, idx, fi, start, stop, cals, mult)
  File "/home/arccha/.conda/envs/pfb/lib/python2.7/site-packages/mne/io/brainvision/brainvision.py", line 174, in _read_segments_c
    dtype = _fmt_dtype_dict[raw.orig_format]
TypeError: unhashable type: 'dict'
```

Am I doing something wrong or is it indeed a bug?

EDIT: I've tested this against MNE 0.15.2 and master branch.

### PR DESCRIPTION
**Pull request by cbrnr**
Fixes #5142.

### CODE REVIEW

**Comment by agramfort** [comment]
@cbrnr can you see why CIs complain?

**Comment by cbrnr** [comment]
Oh yeah, apparently there is a `test_ascii`...

**Comment by agramfort** [comment]
then test is not good enough and we'd better fix the reader as we have
a contract to make this work.

**Comment by cbrnr** [comment]
Agreed. For now I'm raising a `NotImplementedError` only for the case that's actually not implemented since I don't have time to implement it myself right now. If you want to merge this, I could open an issue to remind us to implement this (or maybe someone has time right now, but I think we should do *something* to fix this issue).

**Comment by cbrnr** [comment]
Ready to merge.

File: mne/io/brainvision/brainvision.py [hunk 1]
```diff
@@ -94,11 +94,14 @@ def __init__(self, vhdr_fname, montage=None,
...
-                offsets = np.cumsum([n_skip] + [len(line) for line in f])
-                n_samples = len(offsets) - 1
+                if self._order == 'F':  # multiplexed, channels in columns
+                    n_skip = 0
+                    for ii in range(int(fmt['skiplines'])):
+                        n_skip += len(f.readline())
+                    offsets = np.cumsum([n_skip] + [len(line) for line in f])
+                    n_samples = len(offsets) - 1
+                elif self._order == 'C':  # vectorized, channels, in rows
+                    raise NotImplementedError()
```
Comment (larsoner) [inline mne/io/brainvision/brainvision.py]: This is just a sanity check because it should be caught already by `_get_vhdr_info` right?

Comment (cbrnr) [reply mne/io/brainvision/brainvision.py]: Yes, this is just to remind us where to implement the missing functionality.

Comment (cbrnr) [reply mne/io/brainvision/brainvision.py]: But now that you mention it, the `()` are not really necessary. I guess it is more Pythonic to just `raise NotImplementedError`, right?

**Comment by larsoner** [comment]
Thanks @cbrnr

File: mne/io/brainvision/brainvision.py (hunk 1 above)
Comment (larsoner) [reply mne/io/brainvision/brainvision.py]: The line should never be hit so it doesn't matter too much.

Comment (cbrnr) [reply mne/io/brainvision/brainvision.py]: Yes, it doesn't matter here and actually both variants work.

**Pull request merged**